src/rocks_engine.cpp

```diff
@@ -352,18 +353,18 @@ namespace mongo {
         auto config = _getIdentConfig(ident);
         std::string prefix = _extractPrefix(config);
 
-        RocksRecordStore* recordStore =
+        std::unique_ptr<RocksRecordStore> recordStore =
             options.capped
-                ? new RocksRecordStore(
+                ? stdx::make_unique<RocksRecordStore>(
                       ns, ident, _db.get(), _counterManager.get(), prefix, true,
                       options.cappedSize ? options.cappedSize : 4096,  // default size
                       options.cappedMaxDocs ? options.cappedMaxDocs : -1)
-                : new RocksRecordStore(ns, ident, _db.get(), _counterManager.get(),
+                : stdx::make_unique<RocksRecordStore>(ns, ident, _db.get(), _counterManager.get(),
                                        prefix);
 
         {
             stdx::lock_guard<stdx::mutex> lk(_identObjectMapMutex);
-            _identCollectionMap[ident] = recordStore;
+            _identCollectionMap[ident] = recordStore.get();
```

Comment:
Since a unique pointer is being returned now instead of a raw pointer, I'm a little unsure if this will lead to a dangling pointer in _identCollectionMap after the unique pointer is deallocated. I was unable to find where the pointers in the map are currently deallocated (if at all). Is there some safeguard in place to make sure that the pointers in _identCollectionMap are not used after being deallocated? If not, this change could lead to some issues